app/Services/index_service.py

```diff
@@ -28,11 +32,25 @@ def _prepare_image(self, image: Image.Image, image_data: ImageData, skip_ocr=Fal
             else:
                 image_data.ocr_text = None
 
-    async def index_image(self, image: Image.Image, image_data: ImageData, skip_ocr=False):
+    # currently, here only need just a simple check
+    async def _is_point_duplicate(self, image_data: list[ImageData]) -> bool:
+        image_id_list = [str(item.id) for item in image_data]
+        try:
+            _ = await self._db_context.retrieve_by_ids(image_id_list)
+        except PointNotFoundError:
+            return True  # Only if all points are not found, then it is not duplicate
+        return False
+
+    async def index_image(self, image: Image.Image, image_data: ImageData, skip_ocr=False, allow_overwrite=False):
```

Comment:
In index_image and index_image_batch, there is an `allow_overwrite=False` parameter to enable or disable duplicate checking, when duplicate checking is enabled and duplicates are detected, an exception will be thrown.